pkg/koordlet/util/idleinfo_kidled.go

```diff
@@ -0,0 +1,192 @@
+	Dfea                []uint64 `json:"dfea"`
+	Csua                []uint64 `json:"csua"`
+	Dsua                []uint64 `json:"dsua"`
+	Cfua                []uint64 `json:"cfua"`
+	Dfua                []uint64 `json:"dfua"`
+	Slab                []uint64 `json:"slab"`
+}
+
+func KidledColdPageInfo(path string) (*ColdPageInfoByKidled, error) {
+	data, err := os.ReadFile(path)
+	if err != nil {
+		return nil, err
+	}
+
+	lines := strings.Split(string(data), "\n")
+	statMap := make(map[string]interface{})
+	var info = ColdPageInfoByKidled{}
+	for i, line := range lines {
+		if i == 0 {
+			fields := strings.Fields(line)
+			if len(fields) < 3 {
+				continue
+			}
+			statMap[fields[1][:len(fields[1])-1]] = fields[2]
+		} else if i < 5 {
+			fields := strings.Fields(line)
+			if len(fields) < 3 {
+				continue
+			}
+			val, _ := strconv.ParseUint(fields[2], 10, 64)
+			statMap[fields[1][:len(fields[1])-1]] = val
+		} else if i == 5 {
+			fields := strings.Fields(line)
+			if len(fields) < 3 {
+				continue
+			}
+			s := strings.Split(fields[2], ",")
+			var val = make([]uint64, len(s))
+			for k, v := range s {
+				val[k], _ = strconv.ParseUint(v, 10, 64)
+			}
+			statMap[fields[1][:len(fields[1])-1]] = val
+		} else if i >= 14 {
+			fields := strings.Fields(line)
+			if len(fields) < 1 {
+				continue
+			}
+			var val = make([]uint64, len(fields)-1)
+			for i := 1; i < len(fields); i++ {
+				val[i-1], _ = strconv.ParseUint(fields[i], 10, 64)
+			}
+			statMap[fields[0]] = val
+		}
+	}
+	elem := reflect.ValueOf(&info).Elem()
+	typeOfElem := elem.Type()
+	for i := 0; i < elem.NumField(); i++ {
+		val, ok := statMap[typeOfElem.Field(i).Tag.Get("json")]
+		if ok {
+			if typeOfElem.Field(i).Type.Kind() == reflect.String {
+				elem.Field(i).SetString(val.(string))
+			} else if typeOfElem.Field(i).Type.Kind() == reflect.Uint64 {
+				elem.Field(i).SetUint(val.(uint64))
+			} else if typeOfElem.Field(i).Type.Kind() == reflect.Slice {
+				sliceValue := reflect.ValueOf(val)
+				elem.Field(i).Set(sliceValue)
+			}
+		}
+	}
+	return &info, nil
+}
+
+func (i *ColdPageInfoByKidled) GetColdPageTotalBytes() uint64 {
+	sum := func(nums ...[]uint64) uint64 {
+		var total uint64
+		for _, v := range nums {
+			for _, num := range v {
+				total += num
+			}
+		}
+		return total
+	}
+	return sum(i.Csei, i.Dsei, i.Cfei, i.Dfei, i.Csui, i.Dsui, i.Cfui, i.Dfui, i.Csea, i.Dsea, i.Cfea, i.Dfea, i.Csua, i.Dsua, i.Cfua, i.Dfua, i.Slab)
+}
+
+func (i *ColdPageInfoByKidled) NodeMemWithHotPageUsageBytes() (uint64, error) {
+	Meminfo, err := GetMemInfo()
+	if err != nil {
+		return 0, err
+	}
+	//memWithHotPage=Total-Free-ColdPage
+	memWithHotPageUsageBytes := Meminfo.MemTotal*1024 - Meminfo.MemFree*1024 - i.GetColdPageTotalBytes()
+	return memWithHotPageUsageBytes, nil
+}
+
+func IsKidledSupported() bool {
+	_, err := os.Stat(KidledScanPeriodInSecondsFilePath)
+	if err != nil {
+		klog.Errorf("file scan_period_in_seconds is not exist,err: ", err)
```

Comment:
1. need to format the msg: `,err` -> `, err`
2. should not use the Errorf since a system does not support kidled is not an error